sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java

```diff
@@ -1167,6 +1145,28 @@ private synchronized void handleApiCallSuccess(Object releaseCallId, String rawR
         }
     }
 
+    private synchronized void updateReleaseDetails(ReleaseDetails releaseDetails) {
+        if (mReleaseDownloader != null) {
+
+            /* Cancel previous release downloading. */
+            if (releaseDetails == null || releaseDetails.getId() != mReleaseDownloader.getReleaseDetails().getId()) {
+                mReleaseDownloader.cancel();
+            }
+            mReleaseDownloader = null;
+        }
+        if (mReleaseDownloaderListener != null) {
```

Comment:
Why is this not part of `/* Cancel previous release downloading. */` if? I guess I am missing something.